luigi/worker.py

```diff
@@ -170,14 +170,25 @@ def run(self):
     def _recursive_terminate(self):
         import psutil
 
-        parent = psutil.Process(self.pid)
-        for child in parent.children(recursive=True):
-            child.kill()
+        try:
+            parent = psutil.Process(self.pid)
+            children = parent.children(recursive=True)
+
+            # terminate parent. Give it a chance to clean up
+            super(TaskProcess, self).terminate()
+            parent.wait()
 
-        parent.kill()
+            # terminate children
+            for child in children:
```

Comment:
The reasoning here is that if one child.terminate() fails, we should still try to terminate the others. So the try/except should be inside the loop